chart/templates/deployment.yaml

```diff
@@ -151,11 +151,7 @@ spec:
 {{ toYaml .Values.resources | indent 10 }}
         volumeMounts:
 {{- if .Values.additionalTrustedCAs }}
-        - mountPath: /etc/ssl/certs/ca-additional.pem
-          name: tls-ca-additional-volume
-          subPath: ca-additional.pem
-          readOnly: true
-        - mountPath: /etc/rancher/ssl/ca-additional.pem
```

Comment:
Why are we removing the rancher one entirely ? Going back through old issues I see: 
> The difference in behaviour is the result of a change in the rancher deployment object in the helm chart in v2.2 vs 2.1, in which the custom CA file ca-additional.pem is now mounted at /etc/rancher/ssl/ca-additional.pem vs /etc/ssl/certs/ca-additional.pem (rancher/server-chart@7920ce4#diff-e3ed5b4174b5e628950137770a3e94d0). This change was made to rationalise certificate location into a single, non-OS dependent directory. However, pending updates to the git-based catalog design (15330) this breaks git-based custom catalogs from git servers presenting a cert signed with a custom CA.

So we were setting them in both places on purpose. 

If we do go with this change, it must be in the docs or guides. We'll have to open issue and find/replace.

Reply:
The rancher one is there because we override the certificate location using ```ENV SSL_CERT_DIR /etc/rancher/ssl``` in the image. The other one is for the `git` binary.

Reply:
Now that we are using the SLE image, the [docs](https://documentation.suse.com/suse-caasp/4.5/html/caasp-admin/_troubleshooting_3.html#_error_x509_certificate_signed_by_unknown_authority) indicate that we should use the `/etc/pki/trust/anchors` directory. It seems that `update-ca-certificates` doesn't respect the `SSL_CERT_DIR` environment variable.

All that is to say that since we were setting the `SSL_CERT_DIR` to the custom directory and that is no longer being used, then we can remove it.

Reply:
`SSL_CERT_DIR` is to override the system location for certificates when using Golang, so this can be removed if it is verified that Golang binaries (`rancher` it self but also `rancher-machine` still use the added certificates to validate connections)

Reply:
The `update-ca-certificates` command adds the certs in `/etc/pki/trust/anchors/` to `/etc/ssl/ca-bundle.pem` which is used by Go: https://golang.org/src/crypto/x509/root_linux.go

Reply:
Did you test that docker machine still works with it ? I realize its also golang but we should do a smoke test.

I'm still not sold on this being all thats needed. How do the following still work: 
* Jailer: https://github.com/rancher/rancher/blob/master/package/jailer.sh#L46-L48
* RancherD (though this may be getting removed): https://github.com/rancher/rancher/blob/master/cmd/rancherd/chart/templates/daemonset.yaml#L98-L102
* RKE2: https://github.com/rancher/rancher/blob/master/pkg/controllers/provisioningv2/rke2/machineprovision/template.go#L193-L198